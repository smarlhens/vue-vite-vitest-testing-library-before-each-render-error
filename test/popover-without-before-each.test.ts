import type { RenderResult } from "@testing-library/vue";
import { fireEvent, render } from "@testing-library/vue";
import { afterEach, describe, expect, it } from "vitest";

import Popover from "../src/components/popover.vue";

describe("popover-without-before-each.vue", () => {
  let wrapper: RenderResult;

  afterEach(() => {
    wrapper.unmount();
  });

  it("should render", async () => {
    wrapper = render(Popover);

    const { getByText, html, findByText } = wrapper;
    const button = getByText("Solutions");
    await fireEvent.click(button);
    await findByText("Analytics");
    expect(html()).toMatchSnapshot();
  });
});
