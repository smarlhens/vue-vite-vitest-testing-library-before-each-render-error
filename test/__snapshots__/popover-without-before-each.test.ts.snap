// Vitest Snapshot v1

exports[`popover-without-before-each.vue > should render 1`] = `
"<div data-headlessui-state=\\"open\\" class=\\"relative\\"><button id=\\"headlessui-popover-button-1\\" type=\\"button\\" aria-expanded=\\"true\\" data-headlessui-state=\\"open\\" aria-controls=\\"headlessui-popover-panel-2\\">Solutions</button>
  <!---->
  <transition-stub enteractiveclass=\\"transition duration-200 ease-out\\" enterfromclass=\\"translate-y-1 opacity-0\\" entertoclass=\\"translate-y-0 opacity-100\\" leaveactiveclass=\\"transition duration-150 ease-in\\" leavefromclass=\\"translate-y-0 opacity-100\\" leavetoclass=\\"translate-y-1 opacity-0\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
    <div class=\\"absolute z-10\\" id=\\"headlessui-popover-panel-2\\" data-headlessui-state=\\"open\\">
      <!---->
      <div class=\\"grid grid-cols-2\\"><a href=\\"/analytics\\">Analytics</a><a href=\\"/engagement\\">Engagement</a><a href=\\"/security\\">Security</a><a href=\\"/integrations\\">Integrations</a></div><img src=\\"/solutions.jpg\\" alt=\\"\\">
      <!---->
    </div>
  </transition-stub>
</div>"
`;
